src/vs/base/browser/ui/progressbar/progressbar.ts

```diff
@@ -201,6 +205,8 @@ export class ProgressBar extends Disposable {
 
 	show(delay?: number): void {
 		this.showDelayedScheduler.cancel();
+		this.progressSignal?.dispose();
```

Comment:
going with the default time of 3 seconds here. we can get more feedback to determine if we should use `delay` instead here during May. https://github.com/microsoft/vscode/issues/209661#issuecomment-2060099581